File: Webu/web/app/model/Tree.js
/**
 * Created with IntelliJ IDEA.
 * User: Oleg
 * Date: 20.05.12
 * Time: 10:50
 * To change this template use File | Settings | File Templates.
 */
Ext.define('AM.model.Tree', {
    extend:'Ext.data.Model',
    fields:[
        {name:'key', type:'string'},
        {name:'name', type:'string'},
        {name:'size', type:'int'}
    ],
    idProperty:"key"
})

File: Webu/src/odo/verte/data/Tree.scala
package odo.verte.data

import javax.persistence.{GenerationType, Entity}
import Annotations.{Id, GeneratedValue}
import com.google.appengine.api.datastore.{KeyFactory, Key}
import odo.verte.utils.JSON.map2read


/**
 * Created with IntelliJ IDEA.
 * User: Oleg
 * Date: 20.05.12
 * Time: 18:16
 * To change this template use File | Settings | File Templates.
 */

@Entity
case class Tree(@Id
                @GeneratedValue(strategy = GenerationType.IDENTITY)
                var key: Key,
                var size: Int = 0,
                var name: String = "")

object TreeJSON {
  implicit val serialize = (tree: Tree) => Map(
    "key" -> KeyFactory.keyToString(tree.key),
    "size" -> tree.size,
    "name" -> tree.name
  )

  implicit val deserialize = (map: Map[String, Any]) => Tree(
    key = DB.key(map.read[String]("key")),
    size = map.read[Double]("size").toInt,
    name = map.read[String]("name")
  )
}